pkg/csi/service/common/util.go

```diff
@@ -497,3 +498,35 @@ func ExtractVolumeIDFromPVName(ctx context.Context, volumeName string) (string,
 	}
 	return matches[1], nil
 }
+
+func GetCNSVolumeInfoPatch(ctx context.Context, CapacityInMb int64, volumeId string) (map[string]interface{}, error) {
+	log := logger.GetLogger(ctx)
+	var patch map[string]interface{}
+	if volumeId == "" {
+		log.Errorf("VolumeID %q values cannot be empty", volumeId)
+		return nil, logger.LogNewErrorf(log, "VolumeID values cannot be empty")
+	}
+	if CapacityInMb == -1 {
+		log.Infof("Couldn't retrieve aggregated snapshot capacity for volume %q", volumeId)
+		patch = map[string]interface{}{
+			"spec": map[string]interface{}{
+				"validaggregatedsnapshotsize": false,
+			},
+		}
```

Comment:
Can you add comment here explaning about how this behaves when there is any existing aggregatedSnapshotSizeBytes value and a new one is received at GetCNSVolumeInfoPatch?

how do we deal with concurrency? as in vmsnapshot and volumesnapshot are being taken concurrently?